client/jetpack-cloud/controller.ts

```diff
@@ -206,7 +207,11 @@ export async function cloudSiteSelection(
 	}
 
 	if ( siteFragment ) {
-		if ( context.path.startsWith( '/pricing' ) ) {
+		const languages = getLanguageSlugs().join( '|' );
+		// Regex defines url starts with /:locale/pricing or /pricing
+		const pricingMatchingPath = new RegExp( `(^/(${ languages })/pricing)|^/pricing` );
```

Comment:
A bit nit-picky, but since we don't use the capturing groups, we can optimize it a little by using [non-capturing groups](https://www.regular-expressions.info/brackets.html).

```suggestion
		const pricingMatchingPath = new RegExp( `(?:^/(?:${ languages })/pricing)|^/pricing` );
```